tensorflow/core/framework/op_def_builder.cc

```diff
@@ -114,14 +114,16 @@ bool ConsumeAttrNumber(StringPiece* sp, int64_t* out) {
   } while (false)
 
 bool ConsumeCompoundAttrType(StringPiece* sp, StringPiece* out) {
-  auto capture_data = sp->data();
-  auto capture_begin = sp->begin();
+  const char* capture_data = sp->data();
+  size_t initial_length = sp->length(); 
+
   if (absl::ConsumePrefix(sp, "numbertype") ||
       absl::ConsumePrefix(sp, "numerictype") ||
       absl::ConsumePrefix(sp, "quantizedtype") ||
       absl::ConsumePrefix(sp, "realnumbertype") ||
       absl::ConsumePrefix(sp, "realnumberictype")) {
-    *out = StringPiece(capture_data, sp->begin() - capture_begin);
+    size_t consumed_length = initial_length - sp->length(); // prevent error: 'expression cannot subtract incompatible string_view iterators'
```

Comment:
subtract lengths instead of iterators

Reply:
Are these params in the correct order? They appear to be swapped from the original implementation where it looks like it is doing current - previous and you are doing previous - current.

Reply:
it's a bit confusing but we are calculation the length of the consumed portion of the string using two different approaching.

the original implementation is doing **current position - original position**
I'm doing **initial length - current length**

both work out to the length of the consumed portion

Reply:
I don't really get how that works without taking the absolute value or something, but if you're confident that it's giving you what you want, then it's all good.

Reply:
I'm going to test the logic to make sure it checks out... it's worth double-checking.

Reply:
seems to check out
![image](https://github.com/tausight/tensorflow_custom/assets/2191286/7fa6a48c-e376-413c-8757-50d67758d6a6)